### PR TITLE
Strengthened EmailValidatorTest with RFC boundary condition tests and client-side validation coverage

### PR DESCRIPTION
**Pull request by WarLikeLaux**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

## What does this PR do?

Added 10 new tests to `EmailValidatorTest` to cover RFC boundary conditions and client-side validation methods that had 0% coverage before.

### New tests

- Local part exactly 64 characters (valid per RFC 5321 section 4.5.3.1.1)
- Local part 65 characters (rejected)
- Total address length exactly 254 characters (valid per RFC 2821)
- Total address length 255 characters (rejected)
- Case-insensitive validation
- Default message initialization
- `clientValidateAttribute()` output
- `clientValidateAttribute()` with IDN enabled
- `getClientOptions()` structure and types
- `getClientOptions()` without `skipOnEmpty`

### Before

```
$ vendor/bin/phpunit tests/framework/validators/EmailValidatorTest.php
OK (38 tests, 103 assertions)

$ vendor/bin/phpunit tests/framework/validators/EmailValidatorTest.php --coverage-text --coverage-filter framework/validators/EmailValidator.php
yii\validators\EmailValidator
  Methods:  37.50% ( 3/ 8)   Lines:  61.82% ( 34/ 55)
```

### After

```
$ vendor/bin/phpunit tests/framework/validators/EmailValidatorTest.php
OK (48 tests, 123 assertions)

$ vendor/bin/phpunit tests/framework/validators/EmailValidatorTest.php --coverage-text --coverage-filter framework/validators/EmailValidator.php
yii\validators\EmailValidator
  Methods:  62.50% ( 5/ 8)   Lines:  92.73% ( 51/ 55)
```

### Uncovered lines (4)

- L82: `throw InvalidConfigException` when `enableIDN=true` but intl extension is not installed. Cannot trigger in test environment where intl is present.
- L150-151: `catch ErrorException` in `dns_get_record`. Requires real DNS failure.
- L161: Dead code branch for `PHP_VERSION_ID < 50600`. Yii2 requires PHP >= 7.4.

### Mutation testing

```
$ infection --filter="EmailValidator" --only-covering-test-cases --test-framework-options="--group=validators --exclude-group=db"

100 mutations generated:
      68 mutants were killed
      32 covered mutants were not detected

Mutation Code Coverage: 100%
Covered Code MSI: 68%
```

Undetected mutants are equivalent: DNS network calls, dead code for PHP < 5.6, visibility changes, regex anchor equivalences.

### CODE REVIEW

**Review by terabytesoftw — APPROVED** [review]

File: tests/framework/validators/EmailValidatorTest.php [hunk 1]
```diff
@@ -203,4 +203,131 @@ public function testMalformedAddressesIdnEnabled($value): void
...
+    /**
+     * RFC 2821: total address length must be max 254 characters.
+     */
+    public function testTotalLengthExactly254Characters(): void
+    {
+        $validator = new EmailValidator();
+        $local = str_repeat('a', 64);
+        $domain = str_repeat('a', 63) . '.' . str_repeat('b', 63) . '.' . str_repeat('c', 57) . '.com';
+        $email = $local . '@' . $domain;
+        $this->assertSame(254, strlen($email));
```
Comment (mspirkov) [inline tests/framework/validators/EmailValidatorTest.php]: I don't think it's necessary, as this assertion is always true.

Perhaps the idea is to avoid accidentally breaking the test in the future, but I don't think that makes sense either.

```suggestion
```

File: tests/framework/validators/EmailValidatorTest.php [hunk 2]
```diff
@@ -203,4 +203,131 @@ public function testMalformedAddressesIdnEnabled($value): void
...
+        $this->assertTrue($validator->validate($email));
+    }
+
+    public function testTotalLengthExceeds254Characters(): void
+    {
+        $validator = new EmailValidator();
+        $local = str_repeat('a', 64);
+        $domain = str_repeat('a', 63) . '.' . str_repeat('b', 63) . '.' . str_repeat('c', 58) . '.com';
+        $email = $local . '@' . $domain;
+        $this->assertSame(255, strlen($email));
```
Comment (mspirkov) [inline tests/framework/validators/EmailValidatorTest.php]: I don't think it's necessary, as this assertion is always true.

```suggestion
```

File: framework/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ Yii Framework 2 Change Log
 2.0.55 under development
 ------------------------
 
+- Enh #20744: Strengthen `EmailValidatorTest` coverage (WarLikeLaux)
```
Comment (mspirkov) [inline framework/CHANGELOG.md]: It's not needed here, because source files haven't changed.

```suggestion
```

**Review by mspirkov — CHANGES_REQUESTED** [review]

**Review by mspirkov — APPROVED** [review]

File: tests/framework/validators/EmailValidatorTest.php [hunk 4]
```diff
@@ -203,4 +203,129 @@ public function testMalformedAddressesIdnEnabled($value): void
...
+    /**
+     * RFC 2821: total address length must be max 254 characters.
+     */
+    public function testTotalLengthExactly254Characters(): void
+    {
+        $validator = new EmailValidator();
+        $local = str_repeat('a', 64);
+        $domain = str_repeat('a', 63) . '.' . str_repeat('b', 63) . '.' . str_repeat('c', 57) . '.com';
+        $email = $local . '@' . $domain;
+
```
[the remark this answers is not in the document]
Comment (WarLikeLaux) [reply tests/framework/validators/EmailValidatorTest.php:235]: These assertions were originally present but were removed per reviewer feedback from @mspirkov. The string lengths are deterministic - they are computed from fixed str_repeat() calls and string concatenation, so the assertion is always trivially true and adds no protective value.

File: tests/framework/validators/EmailValidatorTest.php [hunk 5]
```diff
@@ -203,4 +203,129 @@ public function testMalformedAddressesIdnEnabled($value): void
...
+    }
+
+    public function testTotalLengthExceeds254Characters(): void
+    {
+        $validator = new EmailValidator();
+        $local = str_repeat('a', 64);
+        $domain = str_repeat('a', 63) . '.' . str_repeat('b', 63) . '.' . str_repeat('c', 58) . '.com';
+        $email = $local . '@' . $domain;
+
+        $this->assertFalse($validator->validate($email));
```
[the remark this answers is not in the document]
Comment (WarLikeLaux) [reply tests/framework/validators/EmailValidatorTest.php:246]: These assertions were originally present but were removed per reviewer feedback from @mspirkov. The string lengths are deterministic - they are computed from fixed str_repeat() calls and string concatenation, so the assertion is always trivially true and adds no protective value.

**Review by Arhell — COMMENTED** [review]
lgtm

**Comment by samdark** [comment]
Thank you!
